app/models/rules_of_origin/query.rb

```diff
@@ -11,7 +11,9 @@ def initialize(data_set, heading_code, country_code)
     end
 
     def rules
-      id_rules.map(&rule_set.method(:rule))
+      schemes_and_their_id_rules.transform_values do |id_rules|
+        id_rules.map(&rule_set.method(:rule))
+      end
```

Comment:
so, this is returning a hash with the same keys, but the values are
the id_rules mapped to rule_set.method(:rule),
it would be nice to simplify this a bit, or maybe adding a comment 😄

Reply:
Yeh, good should - definitely hard to grok

I've moved the logic for fetching of multiple rules to RuleSet, and added comment in the `transform_values` to explain why we're calling RuleSet from here

Reply:
thanks, I think it's much clear now!